### PR TITLE
Revising the tutorial for usability.

### PR DESCRIPTION
**Pull request by moxiegirl**
- Assume Android Studio environment
- Clarify the SDK and content is pre-release
- Specify expected version of SDK
- Update and clarify the dependency version
- Provide top toc so advanced users can skip ahead
- Test steps and code end-to-end

Signed-off-by: moxiegirl <mary@blockstack.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,189 +1,785 @@
...
+
+1. Choose **Run > Apply changes**.
+2. Choose **Run > Run app** in the emulator.
+3. When you see the application open, choose **Sign in with Blockstack**.
+
+    The system prompts you how to open.
+
+    ![](images/chrome-prompt.png)
+
+4. Choose **Chrome** and click **ALWAYS**.
```
Comment (friedger) [inline README.md]: It also works with Firefox :-)

File: README.md [hunk 2]
```diff
@@ -1,189 +1,785 @@
...
+        private var _blockstackSession: BlockstackSession? = null
+
+
+        override fun onCreate(savedInstanceState: Bundle?) {
+            super.onCreate(savedInstanceState)
+            setContentView(R.layout.activity_main)
+
+            signInButton.isEnabled = false
+
+            val appDomain = URI("https://flamboyant-darwin-d11c17.netlify.com")
```
Comment (friedger) [inline README.md]: add something like
// replace with your app domain

File: README.md [hunk 3]
```diff
@@ -1,189 +1,785 @@
...
+4. Replae the existing the `onCreate` function with the following:
+
+    ```kotlin
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        signInButton.isEnabled = false
+
+        val appDomain = URI("https://flamboyant-darwin-d11c17.netlify.com")
```
Comment (friedger) [inline README.md]: Add comment to change the appDomain

File: README.md [hunk 4]
```diff
@@ -1,189 +1,785 @@
...
 
+    When you are done you should see:
 
-#### Step 3 - Prompt the user to sign in
+    ```JS
+    dependencies {
+        implementation fileTree(dir: 'libs', include: ['*.jar'])
+        implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+        implementation 'com.android.support:appcompat-v7:27.1.0'
+        implementation 'com.android.support.constraint:constraint-layout:1.1.2'
```
Comment (friedger) [inline README.md]: constraint layout is not used in the sample app

File: README.md [hunk 5]
```diff
@@ -1,189 +1,785 @@
...
+5. Set the `defaultConfig minSdkVersion` to `19`.
+
+   When you are done, you should see (within your own username not `moxiegirl`):
+
+   ```JS
+   android {
+       compileSdkVersion 15
+       defaultConfig {
+           applicationId "com.example.moxiegirl.hello_android"
+           minSdkVersion 15
```
Comment (friedger) [inline README.md]: minSdkVersion is 19 as of https://github.com/blockstack/blockstack-android/issues/2

File: README.md [hunk 6]
```diff
@@ -1,189 +1,785 @@
...
+    ```
+
+4. Open the Module `build.gradle` file.
+5. Set the `defaultConfig minSdkVersion` to `19`.
+
+   When you are done, you should see (within your own username not `moxiegirl`):
+
+   ```JS
+   android {
+       compileSdkVersion 15
```
Comment (friedger) [inline README.md]: Compile and Target Sdk Version is 27

File: README.md [hunk 7]
```diff
@@ -1,189 +1,785 @@
...
+### Configure your application with the Blockstack SDK
+
+Now that you have created your initial project and verified it running in an emulator, you are ready to begin configuring the application for use with Blockstack.
+
+1. In studio, open the `AndroidManifest.xml` file.
+2. Add an `<intent-filter>` with the custom handler for Blockstack.
+
+    ```XML
+    <intent-filter>
+      <action android:name="android.intent.action.VIEW" />
```
Comment (friedger) [inline README.md]: the schema element is missing with the custom schema

File: README.md [hunk 8]
```diff
@@ -1,189 +1,785 @@
...
+This tutorial is written for all levels from the beginner to the most
+experienced. For best results, beginners should follow the guide as written. It
+is expected that the fast or furiously brilliant will skip ahead and improvise
+on this material at will. Good speed one and all.
+
+If you prefer, you can skip working through the tutorial all together. Instead,
+you can [download the final project code](images/helloandroid.zip) and import it
+into Android Studio to review it.
+
+## Before you start
```
Comment (friedger) [inline README.md]: Before you start contains only npm, not Android studio or yeoman.

As Android developer, I don't understand why I need npm now.

File: README.md (hunk 1 above)
Comment (moxiegirl) [reply README.md]: Thanks @friedger. Are you seeing Firefox as an option in your emulator? I'm surprised Google allows that to come up in their emulator.

File: README.md (hunk 2 above)
Comment (moxiegirl) [reply README.md]: At this point, I didn't want to do that with a comment, for the tutorial we don't want to do that yet which is why the discourse below the code.

File: README.md (hunk 3 above)
Comment (moxiegirl) [reply README.md]: See the previous comment.  I'm not adding in code comments for the connection info, it was deliberately left as this.  We want them to have a working connection for the first pass.

File: README.md (hunk 5 above)
Comment (moxiegirl) [reply README.md]: Good catch @friedger , cut n paste error. I'll sort that.

File: README.md (hunk 6 above)
Comment (moxiegirl) [reply README.md]: see previous

File: README.md (hunk 7 above)
Comment (moxiegirl) [reply README.md]: Our final example code was simplified.  So, I'm matching what @larrysalibra 's last code was here.

File: README.md (hunk 8 above)
Comment (moxiegirl) [reply README.md]: @friedger good point. I'll sort that. Thanks for the prompt commenting.

File: README.md (hunk 4 above)
Comment (moxiegirl) [reply README.md]: Here, I'm just matching what was in @larrysalibra's final code. I'm going to stet this for now unless his final code includes it.

**Comment by moxiegirl** [comment]
@larrysalibra thanks for the pull request! And I agree with you and @friedger, comments help _in the sample code_ itself.  

Adding comments into the code snippets that _appear in a tutorial discourse_ can be problematic. 

Having comments just in the code snippets and deprecating them in the discussion in the tutorial is often done by novice authors. But, then the author ends up making really long comments in the code **as the purpose** of a tutorial often goes beyond a simple code comment.  

The formatting an author does with text is often referred to as "making things pretty." Formatting isn't done for appearances: it is done for UX.  Proper formatting is key to readability and quick scanning which aids learning.

**Pull request merged**